src/platform/plugins/shared/workflows_management/server/workflows_management/workflows_management_service.test.ts

```diff
@@ -673,6 +673,147 @@ describe('WorkflowsService', () => {
         })
       );
     });
+
+    it('should create workflow with custom ID when provided', async () => {
+      const mockRequest = {
+        auth: {
+          credentials: {
+            username: 'test-user',
+          },
+        },
+      } as any;
+
+      const customId = 'workflow-12345678-abcd-1234-abcd-123456789abc';
+      const workflowCommand = {
+        yaml: 'name: Custom ID Workflow\nenabled: true\ndefinition:\n  triggers: []',
+        id: customId,
+      };
+
+      mockEsClient.search.mockResolvedValue({
+        hits: {
+          total: { value: 0 },
+          hits: [],
+        },
+      } as any);
+      mockEsClient.index.mockResolvedValue({ _id: customId } as any);
+
+      const result = await service.createWorkflow(workflowCommand, 'default', mockRequest);
+
+      expect(result.id).toBe(customId);
+      expect(result.name).toBe('Custom ID Workflow');
+      expect(mockEsClient.index).toHaveBeenCalledWith(
+        expect.objectContaining({
+          id: customId,
+          index: '.workflows-workflows',
+          document: expect.objectContaining({
+            name: 'Custom ID Workflow',
+            enabled: true,
+            createdBy: 'test-user',
+            lastUpdatedBy: 'test-user',
+            spaceId: 'default',
+          }),
+          refresh: 'wait_for',
+          require_alias: true,
+        })
+      );
+    });
+
+    it('should throw WorkflowConflictError when custom ID already exists', async () => {
+      const mockRequest = {
+        auth: {
+          credentials: {
+            username: 'test-user',
+          },
+        },
+      } as any;
+
+      const existingId = 'workflow-12345678-1234-1234-1234-123456789abc';
```

Comment:
Correct me if I'm wrong - looks like this test depends on a workflow created in a previous test, running it alone would probably fail. I'd suggest to make it self contained so tests stay idempotent.